provider/provider.go

```diff
@@ -38,7 +38,9 @@ func New() *schema.Provider {
 			"feature_use_managed_variables": {
 				Type:        schema.TypeBool,
 				Description: "Feature: use managed Terraform variables. If disabled, Terraform variables will be included in legacy Parameter Schema.",
+				Default:     true,
```

Comment:
Future compatibility check: If we ever remove this from the provider, the default=true value won't be propagated and we'll essentially receive a false for this. Can this become a problem, will coder/coder handle it?

Reply:
Once this [PR](https://github.com/coder/coder/pull/8334) is merged, the flag will be ignored. Do you think that we should mention it in the schema description?

Reply:
I think it may be worth a mention 👍🏻

Reply:
I added an extra sentence. Thanks for raising this!